### PR TITLE
Created functionalities in chirp() to add chirps with hashtags

### PR DESCRIPTION
**Pull request by cindyciclta**

### CODE REVIEW

File: src/chirp_client.cc [hunk 1]
```diff
@@ -273,12 +273,37 @@ void ChirpClient::PrintChirpThread(
...
   std::unique_ptr<grpc::ClientReader<chirp::StreamReply>> reader(
       stub_->stream(&context, request));
   // TODO: output the streams here.
   // Sucessfully able to call functions through grpc for streaming hashtags
+  std::vector<chirp::Chirp> chirps;
+  std::string logging_message;
+  // RPC call - read in chirps from stream continously
+  while (true) {
+    if (reader->Read(&reply)) {
+      // chirps.push_back(reply.chirp());
```
Comment (barath) [inline src/chirp_client.cc]: Remove commented code.

File: src/chirp_client.cc [hunk 2]
```diff
@@ -273,12 +273,37 @@ void ChirpClient::PrintChirpThread(
...
+  grpc::Status status = reader->Finish();
+  // Monitoring stopped
+  if (status.ok()) {
+    logging_message = "Streaming ended\n" + status.error_message();
+    std::cout << logging_message << std::endl;
+    LOG(INFO) << logging_message << std::endl;
+    return 0;
+  }
+  // Other
+  else {
```
Comment (barath) [inline src/chirp_client.cc]: Move else up.

File: src/chirp_client.cc [hunk 3]
```diff
@@ -273,12 +273,37 @@ void ChirpClient::PrintChirpThread(
...
+    }
+  }
+
+  grpc::Status status = reader->Finish();
+  // Monitoring stopped
+  if (status.ok()) {
+    logging_message = "Streaming ended\n" + status.error_message();
+    std::cout << logging_message << std::endl;
+    LOG(INFO) << logging_message << std::endl;
+    return 0;
```
Comment (barath) [inline src/chirp_client.cc]: Use "true" and "false" instead of 0 and 1.

File: src/service.cc [hunk 4]
```diff
@@ -20,13 +20,38 @@ bool ServiceLayer::registeruser(const std::string &username) {
   bool success = backend_client_->Put(username, user_serialized);
   return success;
 }
+std::string ServiceLayer::CheckIfHaveHashtag(const std::string &text) {
+
+  std::size_t found = text.find('#');
+  std::string hashtagword = "";
+  if (found != std::string::npos) {
+    int position = static_cast<int>(found);
```
Comment (barath) [inline src/service.cc]: Why the cast?

File: src/service.cc [hunk 5]
```diff
@@ -20,13 +20,38 @@ bool ServiceLayer::registeruser(const std::string &username) {
...
+    for (int i = position; i < text.length(); i++) {
+      if (text[i] != ' ') {
```
Comment (barath) [inline src/service.cc]: This only handles a single space.

File: src/service.cc [hunk 6]
```diff
@@ -20,13 +20,38 @@ bool ServiceLayer::registeruser(const std::string &username) {
...
+        break;
+      }
+    }
+    hashtagword = trim(hashtagword);
+    return hashtagword;
+  }
+}
+std::string ServiceLayer::trim(std::string &str) {
+  size_t first = str.find_first_not_of(' ');
+  if (first == std::string::npos)
```
Comment (barath) [inline src/service.cc]: Braces.

File: src/service.cc [hunk 7]
```diff
@@ -116,8 +141,45 @@ std::optional<chirp::Chirp> ServiceLayer::chirp(const std::string &username,
...
 }
+void ServiceLayer::AddHashtagToDatabase(const chirp::Chirp &chirp,
+                                        const std::string &hashtagword) {
+
+  // Figure out if this hashtag word exist in database already, if so, just
+  // append to that proto, if not make a new hashtag proto
+
+  auto from_get = backend_client_->Get(hashtagword);
+  std::cout << from_get.has_value() << std::endl;
+  if (from_get.value() != "[empty_key]") {
```
Comment (barath) [inline src/service.cc]: Move hard-coded constant string to constant var.

**Comment by cindyciclta** [comment]
Done!!!!

**Review by barath — APPROVED** [review]

File: src/chirp_client.cc [hunk 8]
```diff
@@ -273,12 +273,33 @@ void ChirpClient::PrintChirpThread(
...
+  while (true) {
+    if (reader->Read(&reply)) {
+    }
+  }
+
+  grpc::Status status = reader->Finish();
+  // Monitoring stopped
+  if (status.ok()) {
+    logging_message = "Streaming ended\n" + status.error_message();
+    std::cout << logging_message << std::endl;
```
Comment (apetranik) [inline src/chirp_client.cc:295]: take out cout. probably fine for now while testing

File: src/service.cc [hunk 9]
```diff
@@ -311,4 +372,7 @@ void ServiceLayer::terminate_monitor(chirp::User &user,
 std::vector<chirp::Chirp> ServiceLayer::stream(const std::string hashtag) {
   // TODO add hashtag to database. Create proto to save all the hashtags
   // messages!
+  std::vector<chirp::Chirp> v;
```
Comment (apetranik) [inline src/service.cc:375]: name for this could be more specific

**Review by apetranik — APPROVED** [review]
Looks good! just a small suggestion.

File: src/service.cc (hunk 9 above)
Comment (cindyciclta) [reply src/service.cc:375]: oh, yes I changed it but it will show in the next PR cuz I accidentally added new more stuff on this branch.

**Pull request merged**